R/conditional-transform.R

```diff
@@ -0,0 +1,101 @@
+#' Conditional transformation
+#'
+#' Create a new variable that has values when specific conditions are met. Conditions are specified using a series of formulas: the righthand side is the condition that must be true (a `CrunchLogicalExpr`) and the lefthand side is where to get the value if the condition on the righthand side is true. This is commonly a crunch variable. Imagine that we have two sets of questions, one about what pets one has (Pet1, Pet2, Pet3) and then a second that asks about ones opinion of those pets that one gave in the Pet* series of questions (Opinion1, Opinion2, opinion3). What we ultimately want is what people's opinion of Cats or Dogs is, but because the value of Opinion1, Opinion2, and Opinion3 is dependent on the answer to Pet1, Pet2, and Pet3 respectively, we can't easily get the overall opinions about Cats. `conditionalTransform`  allows you to specify conditions like the following: if variable Pet1 is equal to 'Cat' then use the value from variable Opinion1, if variable Pet2 is equal to 'Cat' then use the value from variable Opinion2, etc. See examples below for how to write this.
+#'
+#' `conditionalTransform` is similar to `makeCaseVariable` however `conditionalTransform` can use other crunch variables as a source of a variable, wehreas, whereas `makeCaseVariable` can only use characters. This additional power comes at a cost: `makeCaseVariable` can be executed entirely on Crunch servers, so no data needs to be downloaded or uploaded to/from the local R session. `conditionalTransform` on the other hand will download the data necesary to construct the new variable.
+#'
+#' @param ... a list of cases to evaluate
+#' @param default_value a default value to use if none of the conditions are true (default: `NA`)
+#' @param type a character that is either "categorical", "text", "numeric" what type of output should be returned?
+#' @param categories if `type="categorical"`, these are all of the categories that should be in the resulting variable
```

Comment:
The first reason was that I was trying to match the feature request. The second was, with the transformations explicit it lets users specify different variable types to use as sources for the transform the use case I imagine most frequently wanting to use both a categorical and a text variable for the source of one of these conditionals, and then being able to specify that it should result in either a text or categorical. This might be over-engineering, but that was what I was aiming for.